libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyUtils.java

```diff
@@ -184,21 +185,79 @@ static List<Entitlement> mergeEntitlements(List<Entitlement> a, List<Entitlement
         return entitlementMap.values().stream().toList();
     }
 
-    static Entitlement mergeEntitlement(Entitlement entitlement1, Entitlement entitlement2) {
-        return switch (entitlement1) {
-            case FilesEntitlement e -> merge(e, (FilesEntitlement) entitlement2);
-            case WriteSystemPropertiesEntitlement e -> merge(e, (WriteSystemPropertiesEntitlement) entitlement2);
-            default -> entitlement1;
+    static Entitlement mergeEntitlement(Entitlement entitlement, Entitlement other) {
+        return switch (entitlement) {
+            case FilesEntitlement e -> mergeFiles(Stream.of(e, (FilesEntitlement) other));
+            case WriteSystemPropertiesEntitlement e -> mergeWriteSystemProperties(Stream.of(e, (WriteSystemPropertiesEntitlement) other));
+            default -> entitlement;
         };
     }
 
-    private static FilesEntitlement merge(FilesEntitlement a, FilesEntitlement b) {
-        return new FilesEntitlement(Stream.concat(a.filesData().stream(), b.filesData().stream()).distinct().toList());
+    public static List<Entitlement> mergeEntitlements(Stream<Entitlement> entitlements) {
```

Comment:
This additional merge logic was a bit of a surprise while I was reviewing this. Do we think this will be imminently useful outside testing? If not, perhaps these utility methods could be moved into `src/test`?

If we do think they belong in production code, they probably deserve their own unit tests.

Reply:
The additional logic is just the generalization from the case of merging 2 (policies/scopes/etc.) to N.
These are used not in tests, but in the creation and display of the plugin installation message.

All this methods are already tested in PolicyUtilsTests.